rpc_util.go

```diff
@@ -900,12 +899,33 @@ func decompress(compressor encoding.Compressor, d mem.BufferSlice, maxReceiveMes
 	//}
 
 	var out mem.BufferSlice
-	_, err = io.Copy(mem.NewWriter(&out, pool), io.LimitReader(dcReader, int64(maxReceiveMessageSize)+1))
+	_, err = io.Copy(mem.NewWriter(&out, pool), io.LimitReader(dcReader, int64(maxReceiveMessageSize)))
```

Comment:
@misvivek can you rebase with latest main branch. This has changed to `mem.ReadAll`. Can you verify if bug still exist?

Reply:
Please verify if the bug still exist first with me.ReadAll. Don't change that